pkg/tsdb/sqleng/sql_engine.go

```diff
@@ -237,86 +173,172 @@ func (e *dataPlugin) DataQuery(ctx context.Context, dsInfo *models.DataSource,
 	return result, nil
 }
 
-// Interpolate provides global macros/substitutions for all sql datasources.
-var Interpolate = func(query plugins.DataSubQuery, timeRange plugins.DataTimeRange, sql string) (string, error) {
-	minInterval, err := interval.GetIntervalFrom(query.DataSource, query.Model, time.Second*60)
-	if err != nil {
-		return sql, nil
+//nolint: staticcheck // plugins.DataQueryResult deprecated
+func (e *dataPlugin) executeQuery(query plugins.DataSubQuery, wg *sync.WaitGroup, queryContext plugins.DataQuery,
+	ch chan plugins.DataQueryResult) {
+	defer wg.Done()
+
+	queryResult := plugins.DataQueryResult{
+		Meta:  simplejson.New(),
+		RefID: query.RefID,
 	}
-	interval := sqlIntervalCalculator.Calculate(timeRange, minInterval)
 
-	sql = strings.ReplaceAll(sql, "$__interval_ms", strconv.FormatInt(interval.Milliseconds(), 10))
-	sql = strings.ReplaceAll(sql, "$__interval", interval.Text)
-	sql = strings.ReplaceAll(sql, "$__unixEpochFrom()", fmt.Sprintf("%d", timeRange.GetFromAsSecondsEpoch()))
-	sql = strings.ReplaceAll(sql, "$__unixEpochTo()", fmt.Sprintf("%d", timeRange.GetToAsSecondsEpoch()))
+	defer func() {
+		if r := recover(); r != nil {
+			e.log.Error("executeQuery panic", "error", r, "stack", log.Stack(1))
+			if theErr, ok := r.(error); ok {
+				queryResult.Error = theErr
+				ch <- queryResult
+			} else if theErrString, ok := r.(string); ok {
+				queryResult.Error = fmt.Errorf(theErrString)
+				ch <- queryResult
+			}
```

Comment:
Shouldn't we deal with the case where `r` is neither `error` nor `string`? In that case, the panic will be swallowed.

Reply:
Was a bit unsure about that. What are you suggesting? `ch <- queryResult` on all panics, but setting `queryResult.Error` if string or error?

Reply:
Since we don't have any clear way to convert other types than string to `error`, I would suggest we either re-panic or assign a generic error to `queryResult.Error`.

Reply:
I'll go with a generic error since re-panic would cause the same issue that I tried to solve with this recovery - not crashing the go-routine. We had that problem with mssql library trying to use a non-supported data type for example if you remember.